runs.py

```diff
@@ -51,9 +50,13 @@ def __init__(self, filename: str, profile: int, data: dict[str, Any]):
         super().__init__(data)
         self.filename = filename
         self.name, _, ext = filename.partition(".")
-        self.matched: dict[str, RunParser] = {}
+        # maybe consider creating a type for this so we can 
+        # say Matched.prev instead of having to know the keys
+        self.matched: dict[str, RunParser] = {} 
         self._character = data["character_chosen"]
         self._profile = profile
+        self._character_streak = None
+        self._rotating_streak = None
```

Comment:
We can use `self._cache["character_streak"]` and `self._cache["rotating_streak"]` for these; instead of checking for None, we just check if the keys are in the dict.